core/trino-main/src/test/java/io/trino/sql/planner/assertions/BasePlanTest.java

```diff
@@ -226,4 +229,44 @@ protected SubPlan subplan(@Language("SQL") String sql, LogicalPlanner.Stage stag
             throw new AssertionError("Planning failed for SQL: " + sql, e);
         }
     }
+
+    protected PlanOptimizer.Context createOptimizerContext(
+            Session session,
+            TypeProvider types,
+            SymbolAllocator symbolAllocator,
+            PlanNodeIdAllocator idAllocator)
+    {
+        return new PlanOptimizer.Context()
+        {
+            @Override
+            public Session getSession()
+            {
+                return session;
+            }
+
+            @Override
+            public TypeProvider getTypeProvider()
+            {
+                return types;
```

Comment:
`getTypeProvider` should be able to return type for symbols allocated via SymbolAllocator

we can go out of sync here.